### PR TITLE
Auth branch

### PR DESCRIPTION
**Pull request by Daniel-Gallagher92**

### CODE REVIEW

File: app/controllers/api/v2/current_user_controller.rb
Comment (fadwil) [inline app/controllers/api/v2/current_user_controller.rb:1]: first time I am seeing `serializable_hash`, is there an advantage to using this over `as_json`? Looking at some of my old code I guess I was using `to_json`, which is just an all-purpose ruby method whereas the other two are both active record specific.

File: app/controllers/users/registrations_controller.rb
Comment (fadwil) [inline app/controllers/users/registrations_controller.rb:1]: I am assuming this is all devise related code?

File: app/controllers/users/sessions_controller.rb
Comment (fadwil) [inline app/controllers/users/sessions_controller.rb:1]: first time seeing the `_opts` parameter for an argument, just looked that up, pretty cool!

File: app/models/catch_lure.rb
Comment (fadwil) [inline app/models/catch_lure.rb:1]: glad to see a catchlure model, not having that relationship felt lacking to me in our v1

File: config/environments/development.rb
Comment (fadwil) [inline config/environments/development.rb:1]: is there a particular reason for this? I havent seen it before but it seems like a best practice in general, in particular to keep an eye out for gem deprecation?

File: config/locales/devise.en.yml
Comment (fadwil) [inline config/locales/devise.en.yml:1]: lots of new stuff for me! did this yaml file get created using `rails generate devise:i18n:views` ?

File: config/routes.rb [hunk 1]
```diff
@@ -1,16 +1,28 @@
...
   # Defines the root path route ("/")
-  # root "articles#index"
+
 
   namespace :api do 
     namespace :v2 do 
-      resources :sessions, only: [:create]
       resources :users, only: [:create] do 
         resources :catches, only: [:create, :index, :show, :update, :destroy]
         resources :lures, only: [:create, :index, :show, :update, :destroy]
```
Comment (fadwil) [inline config/routes.rb:22]: for this and the line above, you are defining all of the RESTful actions so you could refactor to just say

`resources :catches`
`resources :lures`

File: db/migrate/20231129235717_add_jti_to_users.rb
Comment (fadwil) [inline db/migrate/20231129235717_add_jti_to_users.rb:1]: would love to hear more about adding jti to users!

**Review by fadwil — APPROVED** [review]
Good work! I left a handful of comments, feel free to respond here or on slack!

File: app/controllers/api/v2/current_user_controller.rb
Comment (Daniel-Gallagher92) [reply app/controllers/api/v2/current_user_controller.rb:1]: I'm actually learning about this myself, I think I'm going to change it over to to_json or as_json

File: app/controllers/users/registrations_controller.rb
Comment (Daniel-Gallagher92) [reply app/controllers/users/registrations_controller.rb:1]: Yessir

File: app/models/catch_lure.rb
Comment (Daniel-Gallagher92) [reply app/models/catch_lure.rb:1]: Same, I don't know why we didn't use that. At the time it felt so stressful and I guess that speaks to how the mind behaves at the end of Turing lol

File: config/environments/development.rb
Comment (Daniel-Gallagher92) [reply config/environments/development.rb:1]: I'll look more into this, currently responding from the lift line on a powder day

File: config/locales/devise.en.yml
Comment (Daniel-Gallagher92) [reply config/locales/devise.en.yml:1]: I do t recall ever running that specific command, but maybe generated by a similar one.

File: db/migrate/20231129235717_add_jti_to_users.rb
Comment (Daniel-Gallagher92) [reply db/migrate/20231129235717_add_jti_to_users.rb:1]: JTI is short for JWT ID and this column holds the JET generated on new session creation. Upon logout it is destroyed and all of the generating/garbage collection is done by devise in the background

File: config/environments/development.rb
Comment (fadwil) [reply config/environments/development.rb:1]: Heck yeah get it dude!

**Pull request merged**